Geometry_Engine/Query/IsContaining.cs

```diff
@@ -441,6 +441,40 @@ public static bool IsContaining(this PolyCurve curve1, ICurve curve2, bool accep
             return curve1.IsContaining(iPts, acceptOnEdge, tolerance);
         }
 
+        /***************************************************/
+        /**** Public Methods - Cuboid                   ****/
+        /***************************************************/
+
+        public static bool IsContaining(this Cuboid cuboid, IGeometry geometry, bool acceptOnEdge = true, double tolerance = Tolerance.Distance)
+        {
+            TransformMatrix transform = Create.OrientationMatrixLocalToGlobal(cuboid.CoordinateSystem);
+            IGeometry globalGeo = geometry.ITransform(transform);
+            BoundingBox boundingBox = globalGeo.IBounds();
+
+            Point max1 = boundingBox.Max;
+            Point min1 = boundingBox.Min;
+            bool contains = true;
+            if (acceptOnEdge)
+            {
+
+                return ( Math.Abs(min1.X) <= cuboid.Length / 2 + tolerance &&
+                         Math.Abs(max1.X) <= cuboid.Length / 2 + tolerance &&
+                         Math.Abs(min1.Y) <= cuboid.Depth / 2 + tolerance &&
+                         Math.Abs(max1.Y) <= cuboid.Depth / 2 + tolerance &&
+                         Math.Abs(min1.Z) <= cuboid.Height / 2 + tolerance &&
+                         Math.Abs(max1.Z) <= cuboid.Height / 2 + tolerance);
+            }
+            else
+            {
+                return ( Math.Abs(min1.X) < cuboid.Length / 2 + tolerance &&
```

Comment:
Wonder if this should be 
```suggestion
                return ( Math.Abs(min1.X) < cuboid.Length / 2 - tolerance &&
```

instead (if so ofc same for the other rows below if we agree)

Reason being, we are saying we do _not_ accept on edge, and then within a tolerance.

Looking at it like this:

![image](https://user-images.githubusercontent.com/22005920/110000569-bd607d00-7d13-11eb-84c5-9cd3dfa695f8.png)

with the black lines being the tolerances. Then the black lines would be the edge-zone, basically anything within that zone would be "On edge", hence accept on edge should be true for anything within it, but not accepting on edge should exclude it.

Remeber having similar chat with @pawelbaran on some other is containing methods (where I was thinking it should be done as you have done on this PR, but was convinced by pawel that it should be a -tolerance).

Does this make sense? @pawelbaran not sure if you remember the discussion, and if so can confirm?

Reply:
I do not remember the discussion @IsakNaslundBh, but I agree with the suggested changes. I would actually consider going a step further and building a `BoundingBox` based on the input `Cuboid`, then calling `IsContaining` on the former. That could lead to a pattern that could be easily reproduced for other methods implemented for `BoundingBox`.

Reply:
@IsakNaslundBh yes makes sense regarding the subtraction of tolerance when not accepting on edge.
@pawelbaran sorry I don't understand the suggestion of the `BoundingBox`. Can you explain?
Just thinking though... If we create a bounding box on an arbitrarily oriented input cuboid it will be globalCS aligned. Where green is the `BoundingBox` of the red input cuboid below:
![Screenshot 2021-03-04 180024](https://user-images.githubusercontent.com/6618854/110008371-90ac6580-7d13-11eb-9e65-d863f6b2245c.jpg)
I can see how that would work if you mean:
1. create a `BoundingBox` using the cuboid properties around the globalCS
2. then transform the input testGeometry from cuboidCS toGlobal
3. then test box.IsContaining( testGeometry)

Reply:
> I can see how that would work if you mean:
> 
> 1. create a `BoundingBox` using the cuboid properties around the globalCS
> 2. then transform the input testGeometry from cuboidCS toGlobal
> 3. then test box.IsContaining( testGeometry)

Yeah, that is exactly what I had in mind. Not sure how helpful would that be, I can imagine the method becoming a few lines thanks to that:

```
TransformMatrix transform = Create.OrientationMatrixLocalToGlobal(cuboid.CoordinateSystem);
IGeometry globalGeo = geometry.ITransform(transform);
BoundingBox geoBox = globalGeo.IBounds();
BoundingBox cuboidBox = cuboid.BoundingBoxFromProperties();
return cuboidBox.IsContaining(geoBox, tolerance);
```

Then for example `IsInRange` would be limited to a mere copy-paste and change of the last line. Not sure if worth it tbh, it was just an idea that came to my mind.

Reply:
Added in that approach @pawelbaran - much cleaner code. 
I wonder where `BoundingBoxFromProperties` belongs? Any ideas on that? 
Right now as temporary measure I have left it as a private method in the IsContaining file. 
It probably should be public but where logically? It seems like it could be viewed as a `Query`, `Create` or `Convert`? There is risk of confusion if wrongly located / poorly described. 

`IBounds` does not implement a method for cuboids which could add to the confusion. If there was `IBounds` for cuboids it could (and should) return different results from `BoundingBoxFromProperties` when the cuboid is not globally aligned.

Reply:
Based on conversation with @alelom and @al-fisher `BoundingBoxFromProperties` has been refactored as a `Create.BoundingBox` method with appropriate description and parameter names.

Reply:
Hmm, just trying to organize things in my head... How will we calculate the bounding box of a non-aligned `Cuboid` in this case? Through `Bounds` query?

It tempts me to question the above and propose using `Convert` instead of `Create` etc., but it seems that any convention we apply will lead to a certain degree of ambiguity. So makes sense to me, happy with the code now 😉

Reply:
We agree there is some overlap with the Convert - the reason we eventually went for `Create.BoundingBox(Cuboid globallyAlignedCuboid)` is that `Convert` normally expects a more 1:1 conversion. Instead, what this method is doing is creating a new BoundingBox given some properties of a Cuboid; for this new BoundingBox to make sense, you need the Cuboid to be aligned to the global axes, hence why we named the input **`globallyAlignedCuboid`**.